trustmanager/keystore.go

```diff
@@ -218,30 +217,14 @@ func (s *GenericKeyStore) ListKeys() map[string]KeyInfo {
 func (s *GenericKeyStore) RemoveKey(keyID string) error {
 	s.Lock()
 	defer s.Unlock()
-	// If this is a bare key ID without the gun, prepend the gun so the filestore lookup succeeds
-	if keyInfo, ok := s.keyInfoMap[keyID]; ok {
-		keyID = filepath.Join(keyInfo.Gun, keyID)
-	}
-
-	role, legacy, err := getKeyRole(s.store, keyID)
-	if err != nil {
-		return err
-	}
-
 	delete(s.cachedKeys, keyID)
 
-	name := keyID
-	if legacy {
-		name = keyID + "_" + role
-	}
-
 	// being in a subdirectory is for backwards compatibliity
-	err = s.store.Remove(filepath.Join(getSubdir(role), name))
+	err := s.store.Remove(keyID)
 	if err != nil {
 		return err
 	}
 
-	// Remove this key from our keyInfo map if we removed from our filesystem
 	delete(s.keyInfoMap, filepath.Base(keyID))
```

Comment:
is the `filepath.Base` needed anymore here?  (shouldn't hurt anything, but the keyID shouldn't have any more path segments in it)